### PR TITLE
Implement RFC#1006, Deprecate `(action)` and `{{action}}` 

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
RFC: https://github.com/emberjs/rfcs/pull/1006
~~Pending: https://github.com/emberjs/rfcs/pull/1006~~

I was was going through implementing the _removal_ of all this code in a [later PR (for v6)](https://github.com/emberjs/ember.js/pull/20659), I (re?)learned that by deprecating `(action)` and `{{action}}` we shadow-deprecate the `actions: {}` hash in classic.

Should that have its own deprecation RFC?

### CODE REVIEW

File: packages/@ember/-internals/glimmer/lib/helpers/action.ts
Comment (kategengler) [inline packages/@ember/-internals/glimmer/lib/helpers/action.ts:286]: We should use the deprecation url in this format:

`https://deprecations.emberjs.com/id/template-action`

This was implemented here https://github.com/ember-learn/deprecation-app/pull/1261 so we can 1) come up with URLs more easily before merging deprecation guides and 2) Have URLs make sense even if we later decide to extend the `until`

Comment (NullVoxPopuli) [reply packages/@ember/-internals/glimmer/lib/helpers/action.ts]: legit. updated

**Review by phndiaye — APPROVED** [review]

**Comment by NullVoxPopuli** [comment]
Should the available/whatever flags be 5.9 or 5.10?

**Comment by ef4** [comment]
At a first pass the implementation seems good. I guess we should make sure the deprecation URL is live before we merge this.

**Comment by kategengler** [comment]
I agree with @ef4, but syncing that is why we have "Ready for Release". I think this either needs a feature flag or the RFC needs to be in "Ready for Release" to merge.

File: .github/workflows/ci.yml
Comment (kategengler) [inline .github/workflows/ci.yml:24]: This isn't right; further answer in a bit

Comment (NullVoxPopuli) [reply .github/workflows/ci.yml:24]: don't we want the deprecation active in 5.9 (alpha) though?

Comment (kategengler) [reply .github/workflows/ci.yml:24]: No, feature flags are opt-in in alpha and canary builds.

Comment (NullVoxPopuli) [reply .github/workflows/ci.yml:24]: so I should undo all the test changes? (would make doing my other deprecation easier / not be blocked on this PR)

Comment (kategengler) [reply .github/workflows/ci.yml:24]: I hope to unblock this PR today, fwiw.

Comment (NullVoxPopuli) [reply .github/workflows/ci.yml:24]: yay!

File: packages/@ember/-internals/glimmer/tests/integration/components/input-angle-test.js
Comment (kategengler) [inline packages/@ember/-internals/glimmer/tests/integration/components/input-angle-test.js:292]: Should this test remain and be converted to calling a function or is that tested elsewhere?

File: packages/@ember/-internals/glimmer/tests/integration/components/input-angle-test.js
Comment (kategengler) [inline packages/@ember/-internals/glimmer/tests/integration/components/input-angle-test.js:339]: Should this test be converted to using functions?

File: packages/@ember/-internals/glimmer/tests/integration/components/input-curly-test.js
Comment (kategengler) [inline packages/@ember/-internals/glimmer/tests/integration/components/input-curly-test.js:159]: Should this be converted like below?

**Review by kategengler — APPROVED** [review]

File: packages/@ember/-internals/glimmer/tests/integration/components/input-angle-test.js
Comment (NullVoxPopuli) [reply packages/@ember/-internals/glimmer/tests/integration/components/input-angle-test.js:292]: I don't think so, the test title was very clear about testing action behavior?

Comment (kategengler) [reply packages/@ember/-internals/glimmer/tests/integration/components/input-angle-test.js:292]: At some point actions were synonymous with callbacks from the invoked component, so I didn't know if there were any tests that `@enter={{this.foo}}` would call `this.foo`

File: packages/@ember/-internals/glimmer/tests/integration/components/input-angle-test.js
Comment (NullVoxPopuli) [reply packages/@ember/-internals/glimmer/tests/integration/components/input-angle-test.js]: fair enough, updated!

File: packages/@ember/-internals/glimmer/tests/integration/components/input-curly-test.js
Comment (NullVoxPopuli) [reply packages/@ember/-internals/glimmer/tests/integration/components/input-curly-test.js]: fair enough, updated!

**Pull request merged**
